crates/forge/src/fuzzer.rs

```diff
@@ -0,0 +1,171 @@
+use num_traits::One;
+use rand::rngs::StdRng;
+use rand::{Rng, SeedableRng};
+use std::default::Default;
+
+#[derive(Default)]
+#[allow(clippy::module_name_repetitions)]
+pub struct FuzzerRunParams {
+    /// Inclusive value
+    low: BigUint,
+    /// Exclusive value
+    high: BigUint,
+    /// Number of arguments
+    args_number: usize,
+    /// Total number of runs
+    runs_number: u32,
+    /// Number of already executed runs
+    runs_executed: u32,
+    /// Runs in which arguments have min value e.g. `run_with_min_value[0] = 5`
+    /// means that the first argument has the lowest possible value in 5th run
+    runs_with_min_value: Vec<u32>,
+    /// Runs in which arguments have max value e.g. `run_with_min_value[0] = 5`
+    /// means that the first argument has the highest possible value in 5th run
+    runs_with_max_value: Vec<u32>,
+}
+
+#[allow(clippy::module_name_repetitions)]
+pub struct RandomFuzzer {
+    rng: StdRng,
+    fuzzer_run_params: FuzzerRunParams,
+}
+
+impl RandomFuzzer {
+    pub fn new(
+        seed: u64,
+        runs_number: u32,
+        args_number: usize,
+        low: BigUint,
+        high: BigUint,
+    ) -> Self {
+        let rng = StdRng::seed_from_u64(seed);
+
+        let mut fuzzer = RandomFuzzer {
+            rng,
+            fuzzer_run_params: FuzzerRunParams::default(),
+        };
+        fuzzer.set_fuzzer_run_params(runs_number, args_number, low, high);
+
+        fuzzer
+    }
+
+    fn set_fuzzer_run_params(
+        &mut self,
+        runs_number: u32,
+        args_number: usize,
+        low: BigUint,
+        high: BigUint,
+    ) {
+        assert!(low < high);
+        assert!(runs_number >= 3);
+
+        self.fuzzer_run_params.runs_executed = 0;
+        self.fuzzer_run_params.low = low;
+        self.fuzzer_run_params.high = high;
+        self.fuzzer_run_params.runs_number = runs_number;
+        self.fuzzer_run_params.args_number = args_number;
+
+        let runs_with_min_value: Vec<u32> = vec![0; self.fuzzer_run_params.args_number]
+            .into_iter()
+            .map(|_| self.rng.gen_range(1..=self.fuzzer_run_params.runs_number))
+            .collect();
+        let runs_with_max_value: Vec<u32> = vec![0; self.fuzzer_run_params.args_number]
+            .into_iter()
+            .zip(&runs_with_min_value)
+            .map(|(_, &run_with_min)| {
+                let run_with_max = self.rng.gen_range(1..=self.fuzzer_run_params.runs_number);
+                if run_with_max == run_with_min {
+                    run_with_min + 1 % runs_number
+                } else {
+                    run_with_max
+                }
+            })
+            .collect();
+
+        self.fuzzer_run_params.runs_with_max_value = runs_with_max_value;
+        self.fuzzer_run_params.runs_with_min_value = runs_with_min_value;
+    }
+
+    pub fn next_felt252_args(&mut self) -> Vec<Felt252> {
+        assert!(self.fuzzer_run_params.runs_executed < self.fuzzer_run_params.runs_number);
```

Comment:
With this the fuzzer can now fail in runtime if it was misconfigured. I wonder if it shouldn't be handled more gracefully.

Reply:
It is just a sanity check, this check should always pass if the API of fuzzer is used properly by us. I think we should panic if the check fails